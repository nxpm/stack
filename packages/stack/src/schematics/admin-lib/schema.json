{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "AdminLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z]{1}.*$"
    },
    "type": {
      "description": "The type of library",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": {
        "message": "Which type of library would you like to create?",
        "type": "list",
        "items": [
          {
            "value": "assets",
            "label": "Assets library"
          },
          {
            "value": "data-access",
            "label": "Data access library"
          },
          {
            "value": "feature",
            "label": "Feature library"
          },
          {
            "value": "style",
            "label": "Style Library"
          },
          {
            "value": "ui",
            "label": "UI Library"
          },
          {
            "value": "util",
            "label": "Utility library"
          }
        ]
      }
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "publishable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a publishable library."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library."
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    }
  },
  "required": ["name"]
}
