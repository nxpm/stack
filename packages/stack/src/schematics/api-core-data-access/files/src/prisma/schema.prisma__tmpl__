datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id                   String           @id @default(cuid())
  createdAt            DateTime         @default(now())
  updatedAt            DateTime         @default(now()) @updatedAt
  role                 Role
  developer            Boolean          @default(false)
  email                String           @unique
  username             String           @unique
  password             String?
  firstName            String?
  lastName             String?
  avatarUrl            String?
  location             String?
  phone                String?
  bio                  String?
}

enum Role {
  Admin
  User
}

