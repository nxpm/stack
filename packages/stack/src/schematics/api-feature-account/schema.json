{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "ApiFeatureUser",
  "title": "",
  "type": "object",
  "properties": {
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": []
}
