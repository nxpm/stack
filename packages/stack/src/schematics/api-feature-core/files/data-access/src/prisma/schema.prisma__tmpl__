datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  role      Role
  developer Boolean  @default(false)
  username  String   @unique
  password  String?
  firstName String?
  lastName  String?
  avatarUrl String?
  location  String?
  phone     String?
  bio       String?
  emails    Email[]  @relation("Email_belongsTo_Owner")
}

enum Role {
  Admin
  User
}

model Email {
  id            String    @id @default(cuid())
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @default(now()) @updatedAt
  email         String    @unique
  public        Boolean   @default(false)
  primary       Boolean   @default(false)
  verified      Boolean   @default(false)
  verifyToken   String?
  verifyExpires DateTime?
  owner         User      @relation(name: "Email_belongsTo_Owner", fields: [ownerId], references: [id])
  ownerId       String
  // Enable when this is resolved https://github.com/prisma/prisma/issues/3387
  // @@unique([ownerId, primary])
}
