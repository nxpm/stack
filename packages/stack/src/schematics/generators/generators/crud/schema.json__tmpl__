{
  "cli": "nx",
  "id": "web-module",
  "type": "object",
  "properties": {
    "target": {
      "type": "string",
      "description": "Target folder to create the module",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "name": {
      "type": "string",
      "description": "Name of the module (in kebab-case)",
      "$default": {
        "$source": "argv",
        "index": 1
      }
    }
  },
  "required": ["target", "name"]
}
