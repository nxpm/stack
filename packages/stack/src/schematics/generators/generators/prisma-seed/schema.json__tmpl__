{
  "cli": "nx",
  "id": "prisma-seed",
  "type": "object",
  "properties": {
    "confirm": {
      "type": "boolean",
      "description": "Library confirm",
      "x-prompt": "Are you sure you want to delete and seed the database?"
    },
    "timeout": {
      "type": "number",
      "description": "Number of seconds to wait before before starting deletion",
      "default": 10
    }
  },
  "required": ["confirm", "timeout"]
}
