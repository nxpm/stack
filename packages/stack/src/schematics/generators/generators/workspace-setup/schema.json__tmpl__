{
  "cli": "nx",
  "id": "workspace-setup",
  "type": "object",
  "properties": {},
  "required": []
}
