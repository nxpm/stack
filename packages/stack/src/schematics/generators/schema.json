{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "default": "web"
    }
  },
  "required": []
}
