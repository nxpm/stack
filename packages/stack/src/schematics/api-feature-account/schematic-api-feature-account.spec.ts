import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { ApiFeatureAccountSchematicSchema } from './schema'

describe('api-feature-account schematic', () => {
  let appTree: Tree
  const options: ApiFeatureAccountSchematicSchema = { appName: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/api-feature-account', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(
      testRunner.runSchematicAsync('api-feature-account', options, appTree).toPromise(),
    ).resolves.not.toThrowError()

    expect(
      appTree.read('libs/test/account/feature/src/lib/test-account-feature.module.ts').toString(),
    ).toMatchSnapshot()
    expect(
      appTree.read('libs/test/account/feature/src/lib/test-account-feature.resolver.ts').toString(),
    ).toMatchSnapshot()
  })
})
