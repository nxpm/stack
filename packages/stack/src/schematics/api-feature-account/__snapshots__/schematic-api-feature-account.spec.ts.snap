// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api-feature-account schematic should run successfully 1`] = `
"import { Module } from '@nestjs/common'
import { TestAccountDataAccessModule } from '@nxpm/test/account/data-access'

import { TestAccountFeatureResolver } from './test-account-feature.resolver'

@Module({
  imports: [TestAccountDataAccessModule],
  providers: [TestAccountFeatureResolver],
})
export class TestAccountFeatureModule {}
"
`;

exports[`api-feature-account schematic should run successfully 2`] = `
"import { ValidationPipe } from '@nestjs/common'
import { Args, Mutation, Query, Resolver } from '@nestjs/graphql'
import {
  AccountCreateEmailInput,
  AccountUpdatePasswordInput,
  AccountUpdateProfileInput,
  TestAccountDataAccessService,
} from '@nxpm/test/account/data-access'
import { CtxUser } from '@nxpm/test/auth/util'
import { Email, User } from '@nxpm/test/user/data-access'

@Resolver()
export class TestAccountFeatureResolver {
  constructor(private readonly service: TestAccountDataAccessService) {}

  @Query(() => [Email], { nullable: true })
  accountEmails(@CtxUser() user: User) {
    return this.service.accountEmails(user.id)
  }

  @Query(() => User, { nullable: true })
  accountProfile(@CtxUser() user: User) {
    return this.service.accountProfile(user.id)
  }

  @Query(() => Boolean, { nullable: true })
  accountUsernameAvailable(@CtxUser() user: User, @Args('username') username: string) {
    return this.service.accountUsernameAvailable(user.id, username)
  }

  @Mutation(() => Email, { nullable: true })
  accountCreateEmail(@CtxUser() user: User, @Args('input', new ValidationPipe()) input: AccountCreateEmailInput) {
    return this.service.accountCreateEmail(user.id, input)
  }

  @Mutation(() => Email, { nullable: true })
  accountDeleteEmail(@CtxUser() user: User, @Args('emailId') emailId: string) {
    return this.service.accountDeleteEmail(user.id, emailId)
  }

  @Mutation(() => Email, { nullable: true })
  accountMarkEmailPrimary(@CtxUser() user: User, @Args('emailId') emailId: string) {
    return this.service.accountMarkEmailPrimary(user.id, emailId)
  }

  @Mutation(() => Email, { nullable: true })
  accountMarkEmailPrivate(@CtxUser() user: User, @Args('emailId') emailId: string) {
    return this.service.accountMarkEmailPrivate(user.id, emailId)
  }

  @Mutation(() => Email, { nullable: true })
  accountMarkEmailPublic(@CtxUser() user: User, @Args('emailId') emailId: string) {
    return this.service.accountMarkEmailPublic(user.id, emailId)
  }

  @Mutation(() => User, { nullable: true })
  accountUpdateProfile(@CtxUser() user: User, @Args('input', new ValidationPipe()) input: AccountUpdateProfileInput) {
    return this.service.accountUpdateProfile(user.id, input)
  }

  @Mutation(() => User, { nullable: true })
  accountUpdateUsername(@CtxUser() user: User, @Args('username') username: string) {
    return this.service.accountUpdateUsername(user.id, username)
  }

  @Mutation(() => Boolean, { nullable: true })
  accountResetPassword(@CtxUser() user: User) {
    return this.service.accountResetPassword(user.id)
  }

  @Mutation(() => Boolean, { nullable: true })
  accountUpdatePassword(@CtxUser() user: User, @Args('input', new ValidationPipe()) input: AccountUpdatePasswordInput) {
    return this.service.accountUpdatePassword(user.id, input)
  }
}
"
`;
